src/ng/compile.js

```diff
@@ -1215,6 +1259,32 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
 
     var SIMPLE_ATTR_NAME = /^\w/;
     var specialAttrHolder = document.createElement('div');
+
+
+
+    var onChangesTtl = TTL;
+    // The onChanges hooks should all be run together in a single digest
+    // When changes occur, the call to trigger their hooks will be added to this queue
+    var onChangesQueue;
+
+    // This function is called in a $$postDigest to trigger all the onChanges hooks in a single digest
+    function flushOnChangesQueue() {
+      if (!(onChangesTtl--)) {
+        onChangesTtl = TTL;
```

Comment:
I don't think this is correct. We should just `onChangesTtl++`, because the exception will be caught in this digest's `catch` block, so the rest of the ++ (on https://github.com/angular/angular.js/pull/14302/files#diff-a732922b631efed1b9f33a24082ae0dbR1284) will be executed anyway.